pypesto/hierarchical/problem.py

```diff
@@ -0,0 +1,437 @@
+        return inner_problem_from_petab_problem(
+            petab_problem, amici_model, edatas
+        )
+
+    def get_x_ids(self) -> List[str]:
+        """Get IDs of inner parameters."""
+        return list(self.xs.keys())
+
+    def get_xs_for_type(
+        self, inner_parameter_type: str
+    ) -> List[InnerParameter]:
+        """Get inner parameters of the given type."""
+        return [
+            x
+            for x in self.xs.values()
+            if x.inner_parameter_type == inner_parameter_type
+        ]
+
+    def get_dummy_values(self, scaled: bool) -> Dict[str, float]:
+        """
+        Get dummy parameter values.
+
+        Get parameter values to be used for simulation before their optimal
+        values are computed.
+
+        Parameters
+        ----------
+        scaled:
+            Whether the parameters should be returned on parameter or linear
+            scale.
+        """
+        return {
+            x.inner_parameter_id: scale_value(x.dummy_value, x.scale)
+            if scaled
+            else x.dummy_value
+            for x in self.xs.values()
+        }
+
+    def get_for_id(self, inner_parameter_id: str) -> InnerParameter:
+        """Get InnerParameter for the given parameter ID."""
+        try:
+            return self.xs[inner_parameter_id]
+        except KeyError:
+            raise KeyError(f"Cannot find parameter with id {id}.")
+
+    def is_empty(self) -> bool:
+        """Check for emptiness.
+
+        Returns
+        -------
+        ``True`` if there aren't any parameters associated with this problem,
+        ``False`` otherwise.
+        """
+        return len(self.xs) == 0
+
+
+class AmiciInnerProblem(InnerProblem):
+    """
+    Inner optimization problem in hierarchical optimization.
+
+    For use with AMICI objects.
+
+    Attributes
+    ----------
+    edataviews:
+        AMICI ``ExpDataView``s for each simulation condition.
+    """
+
+    def __init__(self, edatas: List[amici.ExpData], **kwargs):
+        super().__init__(**kwargs)
+
+    def check_edatas(self, edatas: List[amici.ExpData]) -> bool:
+        """Check for consistency in data.
+
+        Currently only checks for the actual data values. e.g., timepoints are
+        not compared.
+
+        Parameters
+        ----------
+        edatas:
+            A data set. Will be checked against the data set provided to the
+            constructor.
+
+        Returns
+        -------
+        Whether the data sets are consistent.
+        """
+        data = [
+            amici.numpy.ExpDataView(edata)['observedData'] for edata in edatas
+        ]
+
+        if len(self.data) != len(data):
+            return False
+
+        for data0, data1 in zip(self.data, data):
+            if not np.array_equal(data0, data1, equal_nan=True):
+                return False
+
+        return True
+
+
+def compare_edataviews(edataview0, edataview) -> bool:
```

Comment:
not used anywhere?

Reply:
@dilpath : I removed this for now, as we don't want to deal with the serialization issues at this point.

Reply:
No problem, there's already an alternative in place
https://github.com/ICB-DCM/pyPESTO/blob/9bb6f3ad3ee065ab51c66b817a579693f7934012/pypesto/hierarchical/calculator.py#L89-L94
https://github.com/ICB-DCM/pyPESTO/blob/9bb6f3ad3ee065ab51c66b817a579693f7934012/pypesto/hierarchical/problem.py#L136-L165